### PR TITLE
Fixed bug #148. Added PLAYER_DISCONNECTED event

### PR DESCRIPTION
**Pull request by milutinke**
Sorry for pushing to master, I could not merge it with my copy of the beta branch for some reason, I am not that experienced with git.
#148 is fixed.
Added PLAYER_DISCONNECTED event based on EAC UnregisterClient event.
Also, I did not want to mess anything by installing the packages with yarn and then linting and deleting the packages, so it should be linted.

### CODE REVIEW

File: squad-server/log-parser/player-disconnected.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+export default {
+    regex: /^\[([0-9.:-]+)]\[([ 0-9]*)]LogEasyAntiCheatServer: \[[0-9:]+]\[[A-z]+]\[EAC Server] \[Info]\[UnregisterClient] Client: ([A-z0-9]+) PlayerGUID: ([0-9]{17})/,
+    onMatch: (args, logParser) => {
+        const playerTrackerObjectName = `player-name-tracker-${args[4]}`;
+
+        const data = {
+            raw: args[0],
+            time: args[1],
+            chainID: args[2],
+            playerSuffix: logParser.eventStore[playerTrackerObjectName] ? logParser.eventStore[playerTrackerObjectName] : 'Unknown',
```
Comment (Thomas-Smyth) [inline squad-server/log-parser/player-disconnected.js]: There's probably no need to be working with the player's suffix here (or anywhere in this PR).

The player's suffix is collected using two log events and then is added into their player object listed in the `SquadServer.players` array. As this log line has their SteamID in the `SquadServer` class, where you forward the event, you can use the `.getPlayerBySteamID()` method to get the player's object. Then you can probably remove all other references to the player's suffix from this PR.

**Review by Thomas-Smyth — CHANGES_REQUESTED** [review]
Other than the comment regarding the player's suffix this PR looks good to me.

File: squad-server/log-parser/player-disconnected.js (hunk 1 above)
Comment (milutinke) [reply squad-server/log-parser/player-disconnected.js]: Oh, okay, I did not know it. I will remove that part.

File: squad-server/index.js [hunk 2]
```diff
@@ -178,6 +178,16 @@ export default class SquadServer extends EventEmitter {
       this.emit('PLAYER_CONNECTED', data);
     });
 
+    this.logParser.on('PLAYER_DISCONNECTED', async (data) => {
+      data.player = await this.getPlayerBySteamID(data.steamID);
+      if (data.player) data.player.suffix = data.playerSuffix;
+
+      delete data.steamID;
+      delete data.playerSuffix;
```
Comment (Thomas-Smyth) [inline squad-server/index.js:186]: Looks like you missed these suffix bits.

**Review by Thomas-Smyth — CHANGES_REQUESTED** [review]

File: squad-server/log-parser/player-disconnected.js (hunk 1 above)
Comment (milutinke) [reply squad-server/log-parser/player-disconnected.js]: Alright, done. I've removed the reference to the player suffix.
And I am already using `.getPlayerBySteamID()` in `squad-server/index.js`.

File: squad-server/index.js [hunk 3]
```diff
@@ -178,6 +178,16 @@ export default class SquadServer extends EventEmitter {
...
+
+      this.emit('PLAYER_DISCONNECTED', data);
```
Comment (SeanWalsh95) [inline squad-server/index.js:188]: Would this not always emit nothing as we just deleted the info?

**Comment by milutinke** [comment]
Everything should be alright now. 
Can you merge it with the beta branch or I need to do something more?

**Comment by Thomas-Smyth** [comment]
> Everything should be alright now.
> Can you merge it with the beta branch or I need to do something more?

I don't think you've addressed this yet, or at least not pushed it:
https://github.com/Thomas-Smyth/SquadJS/pull/154#discussion_r585848943

**Pull request merged**